src/ExpressReceiver.ts

```diff
@@ -151,58 +150,52 @@ const respondToUrlVerification: RequestHandler = (req, res, next) => {
   next();
 };
 
-// TODO: this should be imported from another package
-function verifySlackRequest(signingSecret: string): RequestHandler {
-  return async (req , _res, next) => {
-    try {
-      const body: string = (await rawBody(req)).toString();
-      const signature = req.headers['x-slack-signature'] as string;
-      const ts = Number(req.headers['x-slack-request-timestamp']);
-
-      // Divide current date to match Slack ts format
-      // Subtract 5 minutes from current time
-      const fiveMinutesAgo = Math.floor(Date.now() / 1000) - (60 * 5);
-
-      if (ts < fiveMinutesAgo) {
-        const error = errorWithCode(
-          'Slack request signing verification failed. Timestamp is too old.',
-          ErrorCode.ExpressReceiverAuthenticityError,
-        );
-        next(error);
-      }
-
-      const hmac = crypto.createHmac('sha256', signingSecret);
-      const [version, hash] = signature.split('=');
-      hmac.update(`${version}:${ts}:${body}`);
-
-      if (!tsscmp(hash, hmac.digest('hex'))) {
-        const error = errorWithCode(
-          'Slack request signing verification failed. Signature mismatch.',
-          ErrorCode.ExpressReceiverAuthenticityError,
-        );
-        next(error);
-      }
-
-      // Verification passed, assign string body back to request and resume
-      req.body = body;
-      next();
-    } catch (error) {
-      next(error);
-    }
-  };
-}
-
-const parseBody: RequestHandler = (req, _res, next) => {
-  if (req.headers['content-type'] === 'application/x-www-form-urlencoded') {
-    const parsedBody = querystring.parse(req.body);
+/**
+ * Bolt's own Express body-parser.
+ *
+ * This handler parses `req.body` or `req.rawBody`(on Google Could Platform)
+ * and overwrites `req.body` with the parsed JS object.
+ * Following middlewares can expect `req.body` is no long a string value.
+ */
+const parseBody: RequestHandler = async (req, _res, next) => {
+  const stringBody: string = await extractRequestBodyAsString(req);
+  const contentType = req.headers['content-type'];
+  if (contentType === 'application/x-www-form-urlencoded') {
+    const parsedBody = querystring.parse(stringBody);
     req.body = (typeof parsedBody.payload === 'string') ? JSON.parse(parsedBody.payload) : parsedBody;
+  } else if (contentType === 'application/json') {
+    req.body = JSON.parse(stringBody);
   } else {
-    // TODO: should we check the content type header to make sure its JSON here?
-    req.body = JSON.parse(req.body);
+    console.warn(`Unexpected content-type detected: ${contentType}`);
```

Comment:
This is a great idea! Let's try not to use the `console` global from this file, and instead use a `Logger` instance. Since there isn't one available directly here, maybe this sort of concern should be the responsibility of the `ExpressReceiver` class (we can add a `logger` option to the constructor).

Reply:
Thanks for the suggestion. Will do.